### ISSUE TITLE
Getting detected in Firefox even with all other extensions disabled

### ISSUE THREAD

**Issue by ChocoChopin** [0x48piraj/fadblock#148]
![Screenshot 2023-12-04 181415](https://github.com/0x48piraj/fadblock/assets/53260343/5f2ddd22-424d-4acc-8a82-691fb5decc43)
![Screenshot 2023-12-04 181358](https://github.com/0x48piraj/fadblock/assets/53260343/daad4613-55a5-473f-9a5f-2292625f44e8)

This is unfortunate and alarming: it seems that after a long period of having no issues, YouTube is flagging me for using Fadblock, even after ensuring all other browser addons are disabled. Could there be some other cause of this problem? Unfortunately I can't simply switch to Chrome, and I'm afraid of even testing on Chrome lest I reach strike three on my YouTube account. I'm really not sure what to do at this point other than begrudgingly play for premium, having been bullied into it.

**Comment by 0x48piraj** [comment]
You should not pay for anything if you don't want to. Setting that aside, the YT adblocker detection script can't find FadBlock, especially over Firefox in any way ([ref:res-block-attack@0x48piraj](https://github.com/0x48piraj/PwnHouse/tree/master/Privacy/chrome-incognito)) as it does not block any ads, just does active DOM manipulation.

**Comment by scrubborn** [comment]
Mine was also being detected last night, with this being the only youtube / adblock related extension enabled. It seemed to work to block ads on a single video, but when i move to the next video it gives me the warning.

**Comment by youtube-user** [comment]
@0x48piraj 
> Setting that aside, the YT adblocker detection script can't find FadBlock, especially over Firefox in any way ([ref:res-block-attack@0x48piraj](https://github.com/0x48piraj/PwnHouse/tree/master/Privacy/chrome-incognito)) as it does not block any ads, just does active DOM manipulation.

Fadblock is indeed being detected by YouTube. YouTube checks whether the "Skip Ad" button has been clicked programmatically by checking the `isTrusted` attribute of the click event. If `isTrusted` is true, it indicates that the click event was generated by a user action, not by a script or other programmatic means. You can find the event handler on lines 2674 and 2675 of the following script: https://www.youtube.com/s/player/31e0b6d9/player_ias.vflset/en_US/base.js

```js
GG=function(a,b){var c=1,d=[];try{var e=document.querySelector(".ytp-ad-skip-button-slot");e?"none"===getComputedStyle(e).display?d.push("BISCOTTI_BASED_DETECTION_STATE_IS_BUTTON_INVISIBLE"):d.push("BISCOTTI_BASED_DETECTION_STATE_IS_BUTTON_VISIBLE"):d.push("BISCOTTI_BASED_DETECTION_STATE_IS_BUTTON_NOT_FOUND")}catch(f){d.push("BISCOTTI_BASED_DETECTION_STATE_IS_FINDING_BUTTON_FAILURE")}!0===a.isTrusted?d.push("BISCOTTI_BASED_DETECTION_STATE_IS_CLICK_EVENT_TRUSTED"):!1===a.isTrusted?d.push("BISCOTTI_BASED_DETECTION_STATE_IS_CLICK_EVENT_NOT_TRUSTED"):
d.push("BISCOTTI_BASED_DETECTION_STATE_IS_CLICK_EVENT_TRUSTED_UNDEFINED");d.includes("BISCOTTI_BASED_DETECTION_STATE_IS_CLICK_EVENT_NOT_TRUSTED")&&(c=0);wB("ISDSTAT",c);xwa(c,"i.s_",{mca:"sk",metadata:b,states:d});return c};
```

**Comment by 0x48piraj** [comment]
Look at your extensions. As of now, FadBlock still is not getting detected by YouTube. As @ArnoldP-200 told, that's a known issue and it's not a frequent glitch, only happens when the connection drops or the ad loads a bit later than expected.

**Comment by LukeSaward1** [comment]
> Look at your extensions.

Not sure what you're getting at. You can clearly see in the screenshot that Fadblock is the only one enabled, and others are also saying that fadblock is the only extension enabled and yet they're still getting detected.

**Comment by LukeSaward1** [comment]
I'm experiencing the same thing, I guess it needs to be updated or something?

**Comment by JacksonWoodsknight** [comment]
Yeah, I'm getting flags from YouTube saying I'm using an AdBlocker. So it needs to be updated.

**Comment by uthomkirp** [comment]
create fresh profile.  add one extension.

launch profile manager with -P

**Comment by 0x48piraj** [comment]
### Just pushed the new update, 2.0 is coming!

To the loyal existing users, thank you for your support. Simply log in to your accounts, as I have securely migrated all your data, ensuring a seamless transition without the need for account creation or additional payments.

For those who have been using the extension for free, I appreciate your interest in our service. Please note that while you will no longer experience pop-ups, you will need to start a trial account to continue enjoying FadBlock. After the 7-day trial, a lifetime membership will be required to access all features.

Read the [project home](https://github.com/0x48piraj/fadblock) for more information.

Thank you for being part of our journey!

![](https://media.tenor.com/a7vXmLPB_DsAAAAM/ross-rachel.gif)

Closing. Move here: https://github.com/0x48piraj/fadblock/issues/196
